### PR TITLE
Add PgpKey.CreateNewEncryptionSubkey()

### PR DESCRIPTION
**Pull request by paulfurley**

### CODE REVIEW

**Comment by idrysdale** [comment]
Now with added tests! @paulfurley probably best if you review these new tests and if you're happy go ahead and merge!

**Comment by paulfurley** [comment]
First glance looks great, well done! I'll review first thing tomorrow

File: pgpkey/pgpkey_test.go [hunk 1]
```diff
@@ -694,6 +694,37 @@ func TestCreateNewEncryptionSubkey(t *testing.T) {
...
+		outputCipherText := bytes.NewBuffer(nil)
+		w, err := openpgp.Encrypt(
+			outputCipherText,
+			[]*openpgp.Entity{&pgpKey.Entity},
+			&pgpKey.Entity,
+			nil,
+			&config,
+		)
+
+		fmt.Printf("pgpSubkeys length: %d\n", len(pgpKey.Subkeys))
```
Comment (paulfurley) [inline pgpkey/pgpkey_test.go]: I think this is probably a hangover

File: pgpkey/pgpkey_test.go [hunk 2]
```diff
@@ -692,6 +696,17 @@ func TestCreateNewEncryptionSubkey(t *testing.T) {
 
 }
 
+func temporaryWorkAroundSetHashPreference(key *PgpKey) error {
```
Comment (paulfurley) [inline pgpkey/pgpkey_test.go]: Cool, I've linked to this commit in our [trello card](https://trello.com/c/5IfrDbky/185-a-newly-generated-key-doesnt-have-any-cipher-hash-or-compression-preferences) (so let's not do any rebasing or it'll kill the hash)

File: pgpkey/pgpkey_test.go [hunk 3]
```diff
@@ -395,6 +395,16 @@ func TestEncryptionSubkey(t *testing.T) {
 			flagsValid:            true,
 			encryptFlags:          true,
 		},
+		{
```
Comment (paulfurley) [inline pgpkey/pgpkey_test.go]: yay

**Comment by paulfurley** [comment]
Hooray!

File: pgpkey/pgpkey.go [hunk 4]
```diff
@@ -309,6 +310,51 @@ func (key *PgpKey) encryptionSubkey(now time.Time) *openpgp.Subkey {
...
+}
+
+func (key *PgpKey) createNewEncryptionSubkey(validUntil time.Time, now time.Time) error {
+	config := packet.Config{
+		RSABits: 2048,
+	}
+
+	encryptingPriv, err := rsa.GenerateKey(config.Random(), config.RSABits)
+	if err != nil {
+		fmt.Printf("ERROR %v\n", err)
```
Comment (paulfurley) [inline pgpkey/pgpkey.go]: I think this is also a hangover!

**Pull request merged**